src/main/java/net/syscon/elite/service/InmateService.java

```diff
@@ -683,6 +687,24 @@ private void validate(final CategoryRejectionDetail detail) {
         return repository.findInmateAliases(bookingId, defaultOrderBy, sortOrder, offset, limit);
     }
 
+    @VerifyBookingAccess
+    public List<SecondaryLanguage> getSecondaryLanguages(final Long bookingId) {
+        return offenderLanguageRepository
+                .findByOffenderBookId(bookingId)
+                .stream()
+                .filter(lang -> lang.getType().equalsIgnoreCase("SEC"))
+                .map(lang -> SecondaryLanguage
+                        .builder()
+                        .bookingId(lang.getOffenderBookId())
+                        .code(lang.getCode())
+                        .description(lang.getReferenceCode() != null ? lang.getReferenceCode().getDescription() : null)
+                        .canRead(lang.getReadSkill() != null && lang.getReadSkill().equalsIgnoreCase("Y"))
```

Comment:
As above perhaps to get rid of != null check...meh.

`"Y".equalsIgnoreCase(...)`

Reply:
Looks much cleaner, thanks!